### DISCUSSION TITLE
Getting Started with Programming Languages Hub 🗣️

### DISCUSSION THREAD

**Discussion by queenofcorgis** [community/community#51394] [Programming Help]
**General Programming Resources:** 

- [Free Code Camp](https://github.com/freeCodeCamp/freeCodeCamp) - Learn to code for free. 
- [Learn to Program](https://github.com/karlhorky/learn-to-program) - High-quality educational resources, with free content highlighted. 
- [Free Programming Books](https://github.com/EbookFoundation/free-programming-books) - Freely available programming books
- [Mega Project List](https://github.com/karan/Projects) - A list of practical projects that anyone can solve in any programming language
- [Build Your Own](https://github.com/codecrafters-io/build-your-own-x) - This repository is a compilation of well-written, step-by-step guides for re-creating popular technologies from scratch. 
- [Awesome First Pull Request Opportunities](https://github.com/MunGell/awesome-for-beginners) 
- [Practical First Projects](https://github.com/practical-tutorials/project-based-learning) - Offers projects in a variety of languages! 
- [How to Contribute to Bigger Projects](https://github.com/orgs/community/discussions/60447) 

**Programming Languages Threads 🧵**

- [Golang (Go) 🟢 ](https://github.com/orgs/community/discussions/51388)
- [JavaScript ☕ ](https://github.com/orgs/community/discussions/51389)
- [Python 🐍 ](https://github.com/orgs/community/discussions/51390)
- [Ruby ❤️ ](https://github.com/orgs/community/discussions/51392) 

Looking for the Git thread 🔶  - you can find it [here](https://github.com/orgs/community/discussions/51393)!

**Comment by Agestar88** [comment]
The great queen 👑 of world alway give birth to life everyday 🌲🏞️🎋🪴
We learn to move forward

**Comment by PurpleMonkey729** [comment]
Dear Queen, nice to meet you!
You makes our Github richer and richer.
Thanks.
Best wishes!

**Reply by AOHhiroko** [reply]
Sorry for my long absence. Ilack knowledge and do not know what to do.

**Reply by PurpleMonkey729** [reply]
What is the most important is that endless effort.
Please study the trending technologies and make your own creatures.
I think this GitHub will help you.
If you have any question, you can make an issue.
Many excellent developers will help you and answer your question.
This's all.
Best wishes @AOHhiroko !

**Reply by Gigglepit** [reply]
I have a question ❓ I was writing this code or rather chartGPT helped me write it but now whenever I try to run it it says Tronweb is not a constructor what does that mean

**Reply by deebd74** [reply]
لا أعرف البرمجه لكني بحاجه الى المساعده للضروره القسوه

**Comment by DanDevel** [comment]
Learn how to say **'I love you' in #Python** 

heart = '''
     **        **
   ****      ****
 ********  ********
********************
 ******************
   **************
     **********
       *******
         *****
           ***
             *
'''

print(heart)

**Reply by pritykumari1910** [reply]
Thank you for Resources

**Comment by RZ7272** [comment]
> **General Programming Resources:**
> 
> * [Free Code Camp](https://github.com/freeCodeCamp/freeCodeCamp) - Learn to code for free.
> * [Learn to Program](https://github.com/karlhorky/learn-to-program) - High-quality educational resources, with free content highlighted.
> * [Free Programming Books](https://github.com/EbookFoundation/free-programming-books) - Freely available programming books
> * [Mega Project List](https://github.com/karan/Projects) - A list of practical projects that anyone can solve in any programming language
> * [Build Your Own](https://github.com/codecrafters-io/build-your-own-x) - This repository is a compilation of well-written, step-by-step guides for re-creating popular technologies from scratch.
> * [Awesome First Pull Request Opportunities](https://github.com/MunGell/awesome-for-beginners)
> * [Practical First Projects](https://github.com/practical-tutorials/project-based-learning) - Offers projects in a variety of languages!
> 
> **Programming Languages Threads thread**
> 
> * [Golang (Go) green_circle ](https://github.com/orgs/community/discussions/51388)
> * [JavaScript coffee ](https://github.com/orgs/community/discussions/51389)
> * [Python snake ](https://github.com/orgs/community/discussions/51390)
> * [Ruby heart ](https://github.com/orgs/community/discussions/51392)
> 
> Looking for the Git thread large_orange_diamond - you can find it [here](https://github.com/orgs/community/discussions/51393)!

**Comment by tawantan23** [comment]
> **แหล่งข้อมูลการเขียนโปรแกรมทั่วไป:**
> 
> * [Free Code Camp](https://github.com/freeCodeCamp/freeCodeCamp) - เรียนรู้การเขียนโค้ดฟรี
> * [เรียนรู้การเขียนโปรแกรม](https://github.com/karlhorky/learn-to-program) - แหล่งข้อมูลทางการศึกษาคุณภาพสูงพร้อมเนื้อหาฟรีที่เน้นไว้
> * [หนังสือเขียนโปรแกรมฟรี](https://github.com/EbookFoundation/free-programming-books) - หนังสือเขียนโปรแกรมฟรี
> * [Mega Project List](https://github.com/karan/Projects) - รายการโครงการเชิงปฏิบัติที่ใครๆ ก็สามารถแก้ไขได้ในภาษาการเขียนโปรแกรมใดก็ได้
> * [สร้างของคุณเอง](https://github.com/codecrafters-io/build-your-own-x) - พื้นที่เก็บข้อมูลนี้เป็นการรวบรวมคำแนะนำทีละขั้นตอนที่เขียนมาอย่างดีสำหรับการสร้างเทคโนโลยียอดนิยมขึ้นมาใหม่ตั้งแต่ต้น
> * [โอกาสในการขอดึงครั้งแรกที่ยอดเยี่ยม](https://github.com/MunGell/awesome-for-beginners)
> * [โครงการแรกเชิงปฏิบัติ](https://github.com/practical-tutorials/project-based-learning) - เสนอโครงการในหลากหลายภาษา!
> * [วิธีมีส่วนร่วมในโครงการที่ใหญ่กว่า](https://github.com/orgs/community/discussions/60447)
> 
> **หัวข้อภาษาการเขียนโปรแกรม 🧵**
> 
> * [โกลัง (โก) 🟢](https://github.com/orgs/community/discussions/51388)
> * [จาวาสคริปต์ ☕](https://github.com/orgs/community/discussions/51389)
> * [หลาม 🐍](https://github.com/orgs/community/discussions/51390)
> * [รูบี้ ❤️](https://github.com/orgs/community/discussions/51392)
> 
> กำลังมองหาเธรด Git  - คุณสามารถค้นหาได้[ที่นี่](https://github.com/orgs/community/discussions/51393) !

**Comment by rexionmars** [comment]
good work

**Comment by kyuna0312** [comment]
I'm glad you appreciate it! If you'd like to learn more about the Ruby section or have specific questions, feel free to ask. I'm here to help!

**Comment by zmemon** [comment]
Starting out as a new programmer can be overwhelming, but with the right guidance, it can also be an exciting journey. Here’s a beginner’s guide to help you get started:

Choose Your First Programming Language:
Start with a language that's known for being beginner-friendly, like Python. It's widely used and has a clear and readable syntax.
Consider what field interests you—web development (JavaScript, HTML/CSS), data science (Python, R), software development (Java, C#), etc.

Understand the Fundamentals:
Learn the basic concepts such as variables, data types, control structures (if statements, loops), functions, and object-oriented programming.
Practice these concepts by writing small programs.

Set Up Your Environment:

Install the necessary software, like a code editor (Visual Studio Code, Atom), and set up your development environment.
Learn to use version control systems like Git and platforms like GitHub.

Practice Coding Regularly:
Consistency is key. Try to code every day, even if it’s just for 30 minutes.
Work on small projects or coding challenges on platforms like HackerRank, LeetCode, or Codecademy.

Build Projects:
Apply what you’ve learned by building simple projects. This can be as simple as a calculator, a to-do list application, or a personal website.
Completing projects gives you something to show to potential employers and helps consolidate your learning.

Read and Understand Code:
Read other people's code. GitHub is a great place to start.
Try to understand how and why the code works, which will improve your coding skills.

Learn Debugging:
Debugging is an essential skill. Learn to read error messages and use debugging tools to find and fix bugs.

Grasp the Basics of Algorithms and Data Structures:
Start with basic sorting algorithms (like bubble sort, insertion sort), and data structures (like arrays, lists, and dictionaries).

Join a Community:
Engage with communities on platforms like Stack Overflow, Reddit’s r/learnprogramming, or join local meetups.
Being part of a community can provide support, networking opportunities, and mentorship.

Keep Up with the Industry:
Read programming blogs, listen to podcasts, and stay updated with the latest technology news.
Continuous learning is crucial in the ever-evolving field of programming.
Understand Computer Science Fundamentals:

As you get comfortable with programming, start learning computer science basics, including how computers work, operating systems, networks, and databases.

Be Patient and Stay Motivated:
Learning to program takes time and effort. Don’t get discouraged by setbacks.
Celebrate small victories and keep track of your progress.

Remember, every expert programmer was once a beginner. It's a journey of continuous learning and practice. Good luck on your programming adventure!

**Reply by queenofcorgis** [reply]
Hi @zmemon , 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by Muhammad-Rebaal** [comment]
# Embarking on the Exciting Journey of a Novice Programmer

## Introduction

Welcome to the world of programming—a dynamic and ever-evolving realm where creativity meets logic, and innovation knows no bounds. Embarking on the journey as a novice programmer is a thrilling endeavor, filled with endless possibilities and opportunities for growth. Whether you're delving into the intricacies of coding for the first time or transitioning from another field, this new chapter promises a rich and rewarding experience.

## Embracing the Learning Curve

As you set foot on this path, remember that every accomplished programmer was once a beginner. Embrace the learning curve with enthusiasm, as it is through challenges that we sharpen our skills. Programming languages, frameworks, and tools may seem overwhelming at first, but each hurdle overcome is a step closer to mastery.

## Building a Strong Foundation

Just like constructing a sturdy building requires a solid foundation, establishing a strong programming foundation is crucial. Begin by grasping the fundamentals of your chosen language—understand data types, control structures, and basic algorithms. This groundwork will serve as the bedrock upon which you'll build more complex solutions.

## The Power of Collaboration

The programming community is vast and welcoming. Engage in discussions on platforms like [GitHub](https://github.com/), where knowledge is freely shared, and collaboration is encouraged. Participate in open-source projects, contribute to discussions, and learn from experienced developers. Remember, your growth as a programmer is not a solitary journey; it thrives on collaboration and shared expertise.

## Cultivating a Problem-Solving Mindset

At its core, programming is about problem-solving. Approach challenges with a curious and analytical mindset. Break down problems into manageable components, tackle them one step at a time, and celebrate each small victory. The ability to solve problems efficiently is a hallmark of a successful programmer.

## Staying Curious and Adaptable

The tech landscape evolves rapidly, and as a programmer, staying current is essential. Cultivate a curious mindset, always eager to explore new technologies and methodologies. Be adaptable, as the field continually introduces novel paradigms and tools. A willingness to learn and adapt will ensure your skills remain relevant and in demand.

## Celebrating Milestones, No Matter How Small

The journey of a programmer is a series of milestones, both big and small. Whether it's successfully debugging a challenging piece of code, completing your first project, or mastering a new concept, take a moment to celebrate these achievements. Recognizing your progress fuels motivation and reinforces the joy of coding.

## Educational Resources

Here are some educational resources to aid you in your programming journey:

- [Codecademy](https://www.codecademy.com/): Interactive coding lessons across various programming languages.
  
- [freeCodeCamp](https://www.freecodecamp.org/): Free self-paced coding challenges and full-stack development certification.

- [Mozilla Developer Network (MDN)](https://developer.mozilla.org/): Comprehensive web development documentation and tutorials.

- [Coursera](https://www.coursera.org/): Offers online courses in programming, computer science, and related fields from top universities and institutions.

- [edX](https://www.edx.org/): Another platform for online education, providing courses from universities and institutions worldwide.

- [Stack Overflow](https://stackoverflow.com/): A community-driven question and answer platform for programmers to seek help and share knowledge.

## Conclusion

Embarking on the path of a programmer is not just about writing code; it's a journey of continual learning, collaboration, and problem-solving. Embrace the challenges, stay curious, and celebrate your victories. The programming community is here to support you, and your growth is limited only by your imagination and dedication. Happy coding!

**Reply by queenofcorgis** [reply]
Hi @Muhammad-Rebaal , 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by vencomweb** [comment]
Body
Greetings friends I have a small and big problem at the same time I need in a dynamic select to save another value instead of the id in my database for example I select a redip and in the BD I save the name not the id... I would very much appreciate the support provided. I'll leave you the code and the database so you can find your way around

<?php

require 'inc/conexion.php';

$redip = ($_POST['redip']);
$estado = ($_POST['estado']);
$municipio =  ($_POST['municipio']);
$parroquia = ($_POST['parroquia']);
$cuadrante = ($_POST['cuadrante']);
$organismo = ($_POST['organismo']);
$modalidad = ($_POST['modalidad']);
$tipo_manifestacion = ($_POST['tipo_manifestacion']);
$tendencia_politica = ($_POST['tendencia_politica']);
$clasificacion = ($_POST['clasificacion']);
$direccion = ($_POST['direccion']);
$peticiones = ($_POST['peticiones']);
$ente_adscrito = ($_POST['ente_adscrito']);
$lider_marcha = ($_POST['lider_marcha']);
$status = ($_POST['status']);
$cantidad_personas = ($_POST['cantidad_personas']);
$turno = ($_POST['turno']);
$hora_inicio = ($_POST['hora_inicio']);
$hora_final = ($_POST['hora_final']);
$detenidos = ($_POST['detenidos']);
$lesionados = ($_POST['lesionados']);
$fallecidos = ($_POST['fallecidos']);
$observaciones = ($_POST['observaciones']);
$imagen_concentracion = ($_POST['imagen_concentracion']);
$jerarquia = ($_POST['jerarquia']);
$apellidos_nombres = ($_POST['apellidos_nombres']);
$celular = ($_POST['celular']);
$dependencia_adscrito = ($_POST['dependencia_adscrito']);
$servicio_adscrito = ($_POST['servicio_adscrito']);

//$idredip = mysqli_query($mysqli, "SELECT redip FROM redip WHERE id_redip = '$redip'");
//$idestado = mysqli_query($mysqli, "SELECT estado FROM estado WHERE id_estado = '$estado'");
//$idmunicipio = mysqli_query($mysqli, "SELECT municipio FROM municipio WHERE id_municipio = '$municipio'");
//$idparroquia = mysqli_query($mysqli, "SELECT parroquia FROM parroquia WHERE id_parroquia = '$parroquia'");
//$idcuadrante = mysqli_query($mysqli, "SELECT cuadrante FROM cuadrante WHERE id_cuadrantePaz = '$cuadrante'");
//$idorganismo = mysqli_query($mysqli, "SELECT organismo FROM organismo WHERE id_organismo = '$organismo'");

$sql = "INSERT INTO manifestaciones (fecha,	redip, estado, municipio, parroquia, cuadrante, organismo, modalidad, tipo_manifestacion, tendencia_politica, clasificacion, direccion, peticiones, ente_adscrito, lider_marcha, status, cantidad_personas, turno, hora_inicio, hora_final, detenidos, lesionados, fallecidos, observaciones, imagen_concentracion,  jerarquia, apellidos_nombres, celular, dependencia_adscrito, servicio_adscrito) 
VALUES (NOW(), '$redip', '$estado', '$municipio', '$parroquia', '$cuadrante', '$organismo', '$modalidad', '$tipo_manifestacion', '$tendencia_politica', '$clasificacion', '$direccion', '$peticiones', '$ente_adscrito', '$lider_marcha', '$status', '$cantidad_personas', '$turno', '$hora_inicio', '$hora_final', '$detenidos', '$lesionados','$fallecidos','$observaciones','$imagen_concentracion', '$jerarquia', '$apellidos_nombres', '$celular', '$dependencia_adscrito', '$servicio_adscrito')";

if ($mysqli->query($sql)) {
    $id = $mysqli->insert_id;
    
}

header('location: ../login/inicio.php');
$mysqli->close();

// Para guardar los valores en la tabla

// Busca los IDs correspondientes en la base de datos
//$idredip = mysqli_query($conn, "SELECT redip FROM redip WHERE id_redip = '$redip'");
//$idmodelo = mysqli_query($conn, "SELECT idmodelo FROM modelo WHERE modelo = '$modelos'");
//$idcolor = mysqli_query($conn, "SELECT idcolor FROM color WHERE color = '$color'");

// Inserta los valores en la tabla
//$sql = "INSERT INTO reservas (marca, modelo, color) VALUES ('$idredip', '$idmodelo', '$idcolor')";
//if ($conn->query($sql) === TRUE) {
//    header('Location: ../index.php');
//} else {
//    echo "Error: " . $sql . "<br>" . $conn->error;
//}
//$conn->close();

?>

**Comment by AdityaSrivastava185** [comment]
Hello , can i get to know that i am looking for people who can review my project amd contribute to it , where should i post my project for further developments , The project is reactjs-for-beginners and It aims to be a resource for learning reactjs and a starting point for all the beginners contributors to get started with open source , So currently looking forward to improve the docs of the project and also add some new stuffs and raise issue for the [contribution](https://github.com/AdityaSrivastava185/Reactjs-For-Beginners)

**Comment by goldentalent626** [comment]
I think you are very excellent developer.
Thanks

**Comment by AdityaSrivastava185** [comment]
Hello , I want to start building a full stack project but i am confused between the tech stacks , should i choose mern stack or build with (mongodb , express , nodejs and ejs ) stack to build the application , and also that if i would build with ejs template engine is it still going to help me in future like mern stack or not , is choosing ejs a good option ?? Please help me to solve this issue

**Comment by goldentalent626** [comment]
I think mern stack is the best option.
As you know, React is very popular javascript library and It is using virtual DOM.
So It has fast speed.
That's all.

**Comment by goldentalent626** [comment]
hi

**Comment by sanchezta** [comment]
Super useful

**Comment by aziz-codes** [comment]
If you are excited about web development then you can go with initially HTML/CSS and JS, then you pick Angular or React, Angular required prior knowledge of Typescript, so React is a good choice, then Next JS and then you can go with backend like Node, express and some other frameworks.

Try to build real time projects you will get a better idea.

**Comment by Shashwenth** [comment]
# **Starting Your Programming Journey**
If you're new to programming, I highly recommend starting with C. C is a foundational language that gives you full control over your code, allowing you to understand the underlying mechanics of how your programs work. It's like having a remote control to your code, where it executes exactly what you instruct it to do, no more, no less. Once you're comfortable with C, you can explore other programming languages depending on the specific field or industry you're interested in. I suggest using platforms like LinkedIn to connect with professionals in your area of interest and gain insight into the most relevant languages for your career goals.

## Key Learning Resources
### Data Structures & Algorithms
To build a solid foundation in programming, start by mastering data structures. FreeCodeCamp offers a comprehensive and free curriculum that covers both basic and advanced concepts. Understanding how to build and manipulate custom data structures will significantly improve your problem-solving skills.

### Algorithmic Thinking
Once you have a strong grasp on data structures, take the next step by diving into algorithms. Platforms like Leetcode provide an excellent environment for practicing algorithmic challenges. This will help you develop the critical thinking skills necessary to determine how, when, and which algorithm to apply to a given problem.

### Practical Application: Build Projects
Building projects >>> YouTube. Youtube will not make you a proficient programmer—just as you can't learn to play the piano by watching others play. The key is consistent practice and building real-world projects that challenge and expand your skills.

 
**If you are unsure about which project to take on, check out codingchallenges.fyi . It has a variety of problems ranging in different areas of concentrations.**

**_Best Of Luck._**

**Comment by Manhhatthien98** [comment]
> **Tài nguyên lập trình chung:**
> 
> * [Free Code Camp](https://github.com/freeCodeCamp/freeCodeCamp) - Học lập trình miễn phí.
> * [Học lập trình](https://github.com/karlhorky/learn-to-program) - Nguồn tài nguyên giáo dục chất lượng cao, có nội dung miễn phí nổi bật.
> * [Sách lập trình miễn phí](https://github.com/EbookFoundation/free-programming-books) - Sách lập trình miễn phí
> * [Mega Project List](https://github.com/karan/Projects) - Danh sách các dự án thực tế mà bất kỳ ai cũng có thể giải quyết bằng bất kỳ ngôn ngữ lập trình nào
> * [Tự xây dựng](https://github.com/codecrafters-io/build-your-own-x) - Kho lưu trữ này là tập hợp các hướng dẫn từng bước được viết rõ ràng để tái tạo các công nghệ phổ biến từ đầu.
> * [Cơ hội tuyệt vời cho yêu cầu kéo đầu tiên](https://github.com/MunGell/awesome-for-beginners)
> * [Dự án thực tế đầu tiên](https://github.com/practical-tutorials/project-based-learning) - Cung cấp các dự án bằng nhiều ngôn ngữ khác nhau!
> * [Làm thế nào để đóng góp vào các dự án lớn hơn](https://github.com/orgs/community/discussions/60447)
> 
> **Ngôn ngữ lập trình Threads 🧵**
> 
> * [Golang (Đi) 🟢](https://github.com/orgs/community/discussions/51388)
> * [JavaScript ☕](https://github.com/orgs/community/discussions/51389)
> * [Trăn 🐍](https://github.com/orgs/community/discussions/51390)
> * [Hồng ngọc ❤️](https://github.com/orgs/community/discussions/51392)
> 
> Đang tìm kiếm chủ đề Git 🔶 - bạn có thể tìm thấy nó [ở đây](https://github.com/orgs/community/discussions/51393) !

**Comment by lokmanTahata** [comment]
I wonder if any of you around here are self taught , and if so how did you get your first job ? And how much time it took you to do it?

**Comment by showhand168** [comment]
it 's very userful for me.thanks.i am super ideo from wanshen."this this this ,突次"

**Comment by kateyohannes** [comment]
i use this configuations and setp. try it, it may help
# Setp 1: add custom configuration to handle client-side routing
`server {
    listen 80;
    server_name yourdomain.com;

    location / {
        root /usr/share/nginx/html;
        try_files $uri /index.html;
    }

    location /static/ {
        root /usr/share/nginx/html;
    }

    location /api/ {
        proxy_pass http://backend:5000;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}`

# Sept 2: update your docker file: for example
### Stage 1: Build the React app
`
FROM node:16 AS build
WORKDIR /app
COPY package*.json ./
RUN npm install
COPY . .
RUN npm run build
`
### Stage 2: Serve the app with Nginx
`
FROM nginx:1.26.2
COPY --from=build /app/build /usr/share/nginx/html
COPY nginx.conf /etc/nginx/conf.d/default.conf
EXPOSE 80
CMD ["nginx", "-g", "daemon off;"]
`

# Step 3: to run and build docker image: docker-compose.yaml
`version: '3.8'
services:
  web:
    image: your-react-app
    build:
      context: .
      dockerfile: Dockerfile
    ports:
      - "80:80"
    restart: always`

# Step 4: finally, rebuild and run the container
`docker compose build
docker compose up -d`

**Reply by smartberry1122** [reply]
> i use this configuations and setp. try it, it may help
> 
> # Setp 1: add custom configuration to handle client-side routing
> `server { listen 80; server_name yourdomain.com;
> 
> ```
> location / {
>     root /usr/share/nginx/html;
>     try_files $uri /index.html;
> }
> 
> location /static/ {
>     root /usr/share/nginx/html;
> }
> 
> location /api/ {
>     proxy_pass http://backend:5000;
>     proxy_set_header Host $host;
>     proxy_set_header X-Real-IP $remote_addr;
>     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
>     proxy_set_header X-Forwarded-Proto $scheme;
> }
> ```
> 
> }`
> 
> # Sept 2: update your docker file: for example
> ### Stage 1: Build the React app
> `FROM node:16 AS build WORKDIR /app COPY package*.json ./ RUN npm install COPY . . RUN npm run build`
> 
> ### Stage 2: Serve the app with Nginx
> `FROM nginx:1.26.2 COPY --from=build /app/build /usr/share/nginx/html COPY nginx.conf /etc/nginx/conf.d/default.conf EXPOSE 80 CMD ["nginx", "-g", "daemon off;"]`
> 
> # Step 3: to run and build docker image: docker-compose.yaml
> `version: '3.8' services: web: image: your-react-app build: context: . dockerfile: Dockerfile ports: - "80:80" restart: always`
> 
> # Step 4: finally, rebuild and run the container
> `docker compose build docker compose up -d`

**Comment by prabhatadvait** [comment]
Hello GitHub Community, 👋

I’m Prabhat Kumar, a 3rd-year Data Science student with a good foundation in Pandas and NumPy. I’m looking for advice on advancing my skills and preparing for real-world applications and interviews.

Best Learning Approach:

Should I focus on building a project that covers most Pandas/NumPy operations or solve specific manipulation problems for revision?
What advanced topics or techniques should I focus on for interviews?
Real-World Readiness:

Can you recommend datasets, case studies, or interview-style exercises for mastering these libraries?
Any resources or workflows you’d suggest for hands-on practice?
I’d appreciate any guidance or tips to make my learning efficient and practical! 🙏

**Comment by jaiganesh2108** [comment]
Thank you for putting this together! This is a fantastic starting point for anyone beginning their programming journey. Having all these resources and language-specific threads in one place makes it much easier to get started. I really appreciate the effort that went into organizing this.
